lib/read_only/read_only_base.jinja

```diff
@@ -0,0 +1,365 @@
+{# Added this as jinja code so that it is understood that the
+   comment does not apply to THIS file. #}
+{{ "! ==================================================" }}
+{{ "! THIS FILE IS CREATED FROM THE JINJA TEMPLATE FILE " }}
+{{ "! DO NOT MODIFY DIRECTLY                            " }}
+{{ "! ==================================================" }}
+
```

Comment:
Please add a comment here giving an overview of this file. e.g. "Jinja-templated implementation of XXX. Jinja templates are used to YYY"

Reply:
Done.

Reply:
Great, thanks.